include/substrait-mlir/Dialect/Substrait/IR/SubstraitTypes.td

```diff
@@ -75,6 +75,15 @@ def Substrait_TimestampTzType : Substrait_Type<"TimestampTz", "timestamp_tz"> {
   }];
 }
 
+/// Currently supported atomic types. These correspond directly to the types in
+def Substrait_UUIDType : Substrait_Type<"UUID", "uuid"> {
+  let summary = "Substrait uuid type";
+  let description = [{
+    This type represents a substrait uuid (universally-unique identifier) type.
+    This is a 16-byte binary value.
```

Comment:
I am wondering how exactly the values are represented. I *suppose* that the string is used to represent the hexadecimal format of the 128-bit value: `c48ffa9e-64f4-44cb-ae47-152b4e60e77b`. It's important to say that -- it could also represent the value as a 16-byte string as is.

We might also think whether we don't want to use that representation. It corresponds to (what I think is) the representation of literals in protobuf plans, so it would make import and export easy.

In any case, I think it would be good to add a `def UUIDAttr` with a constrained attribute allowing only 16-byte values. (In the latter of the two cases above, that should be easy: simply a length of 16. In the former, we would probably have use some regex to make sure we only have valid values.)

Reply:
We *could* also represent the value internally as an `i128`; however, that'd require us to convert to and from 16-byte arrays, which I am not sure how easy it is. We can also leave that as a potential future optimization.